x-pack/plugins/observability_solution/inventory/public/components/search_bar/entity_types_controls.tsx

```diff
@@ -61,7 +59,7 @@ export function EntityTypesControls({ onChange }: Props) {
       options={options}
       selectedOptions={selectedOptions}
       onChange={(newOptions) => {
-        onChange(newOptions.map((option) => option.key as EntityType));
+        onChange(newOptions.map((option) => option.key!));
```

Comment:
You could remove the `!` if you do this. It might be safer if, for some reason, `key` gets to be `undefined`.

```ts
newOptions.map((option) => option.key).filter((key): key is string => !!key)
```